lib/msf/core/exploit/remote/ms_samr.rb

```diff
@@ -0,0 +1,271 @@
+    samr_con.samr.close_handle(samr_con.server_handle) if samr_con.server_handle
+  end
+
+  def lookup_computer(opts = {})
+    tree = opts[:tree] || connect_ipc
+
+    samr_con = connect_samr(tree)
+
+    computer_name = opts[:computer_name] || datastore['COMPUTER_NAME']
+    if computer_name.blank?
+      raise MsSamrBadConfigError, 'Unable to lookup the computer account since its name is unknown'
+    end
+    details = samr_con.samr.samr_lookup_names_in_domain(domain_handle: samr_con.domain_handle, names: [ computer_name ])
+    if details.nil?
+      raise MsSamrBadConfigError, 'The specified computer was not found.'
+    end
+    details = details[computer_name]
+    sid = samr_con.samr.samr_rid_to_sid(object_handle: samr_con.domain_handle, rid: details[:rid]).to_s
+    print_good("Found #{samr_con.domain_name}\\#{computer_name} (SID: #{sid})")
+  ensure
+    samr_con.samr.close_handle(samr_con.domain_handle) if samr_con.domain_handle
+    samr_con.samr.close_handle(samr_con.server_handle) if samr_con.server_handle
+  end
+
+
+  module_function
+
+  def connect_ipc
+    begin
+      connect
+    rescue Rex::ConnectionError => e
+      raise MsSamrConnectionError, e.message
+    end
+
+    begin
+      smb_login
+    rescue Rex::Proto::SMB::Exceptions::Error, RubySMB::Error::RubySMBError => e
+      raise MsSamrAuthentcationError, "Unable to authenticate ([#{e.class}] #{e})."
+    end
+    report_service(
+      host: rhost,
+      port: rport,
+      host_name: simple.client.default_name,
+      proto: 'tcp',
+      name: 'smb',
+      info: "Module: #{fullname}, last negotiated version: SMBv#{simple.client.negotiated_smb_version} (dialect = #{simple.client.dialect})"
+    )
+
+    begin
+      simple.client.tree_connect("\\\\#{sock.peerhost}\\IPC$")
+    rescue RubySMB::Error::RubySMBError => e
+      raise MsSamrConnectionError, "Unable to connect to the remote IPC$ share ([#{e.class}] #{e})."
+    end
+  end
+
+  def connect_samr(tree)
+    begin
+      vprint_status('Connecting to Security Account Manager (SAM) Remote Protocol')
+      samr = tree.open_file(filename: 'samr', write: true, read: true)
+
+      vprint_status('Binding to \\samr...')
+      samr.bind(endpoint: RubySMB::Dcerpc::Samr)
+      vprint_good('Bound to \\samr')
+      server_handle = samr.samr_connect
+    rescue RubySMB::Dcerpc::Error::FaultError => e
+      elog(e.message, error: e)
+      raise MsSamrUnexpectedReplyError, "Connection failed (DCERPC fault: #{e.status_name})"
+    end
+
+    if datastore['SMBDomain'].blank? || datastore['SMBDomain'] == '.'
+      all_domains = samr.samr_enumerate_domains_in_sam_server(server_handle: server_handle).map(&:to_s).map(&:encode)
+      all_domains.delete('Builtin')
+      if all_domains.empty?
+        fail_with(Failure::NotFound, 'No domains were found on the SAM server.')
+        raise MsSamrNotFoundError, 'No domains were found on the SAM server.'
```

Comment:
Looks like this might be a duplicate

```suggestion
        raise MsSamrNotFoundError, 'No domains were found on the SAM server.'
```